### PR TITLE
Tutorial - Add basic framework

### PR DESCRIPTION
**Pull request by Spoffy**
Adds support for CfgHints entries to show as a survival card hint, when triggered by one of these conditions:
- Entering a specific marker on the map
- The client receiving a specific event from any source
- Looking at an object with `vgm_c_tutorial` set to the path of the tutorial.

### CODE REVIEW

File: game/functions/systems/tutorial/client/fn_tutorial_postInit.sqf [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        private _triggerConfig = _hintConfig >> "Triggers";
+        private _marker = getText (_triggerConfig >> "markerEntered");
+        if (_marker != "" && markerPos _marker isNotEqualTo [0, 0, 0]) then {
+            _marker setMarkerAlphaLocal 0;
+            private _trigger = createTrigger ["EmptyDetector", markerPos _marker, false];
+            _trigger setTriggerActivation ["ANYPLAYER", "PRESENT", true];
+            private _markerSize = markerSize _marker;
+            _trigger setTriggerArea [_markerSize # 0, _markerSize # 1, markerDir _marker, markerShape _marker isEqualTo "RECTANGLE"];
+            _trigger setTriggerStatements [
+                "this && player in thisList",
```
Comment (veteran29) [inline game/functions/systems/tutorial/client/fn_tutorial_postInit.sqf]: Lazy eval would be slower here most likely so we can simply omit `this`.
```suggestion
                "player in thisList",
```

File: game/functions/systems/tutorial/client/fn_tutorial_preInit.sqf [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    Author:
+    Date: 2024-11-16
+    Last Update: 2024-11-17
+    Public: No
+
+    Description:
+        PreInit for the tutorial system
+
+*/
+
```
Comment (veteran29) [inline game/functions/systems/tutorial/client/fn_tutorial_preInit.sqf:14]:
```suggestion

if (!hasInterface) exitWith {};
```

File: game/functions/systems/tutorial/client/fn_tutorial_preInit.sqf [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    createHashMapFromArray [
+        ["name", "TutorialAcknowledgeHint"],
+        ["displayName", "STR_VGM_TUTORIAL_UI_ACKNOWLEDGE_HINT"],
+        ["onRelease", false],
+        ["defaultKey", createHashMapFromArray [
+            ["dikCode", DIK_8]
+        ]]
+    ]
+] call para_c_fnc_keyhandler_registerAction;
+["TutorialAcknowledgeHint", para_c_fnc_ui_hints_acknowledgeHintKeypress] call para_c_fnc_keyhandler_addGeneralActionHandler;
```
Comment (veteran29) [inline game/functions/systems/tutorial/client/fn_tutorial_preInit.sqf:28]: I feel like we should re-use vanilla `help` action for this.
<https://community.bistudio.com/wiki/addUserActionEventHandler>
<https://community.bistudio.com/wiki/inputAction/actions>
`help` is the action name for vanilla Adv Hints.

File: game/functions/systems/tutorial/client/fn_tutorial_trigger.sqf [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    Example(s):
+        ["Tutorial", "Tutorial1"] call vgm_c_fnc_tutorial_trigger;
+ */
+
+params ["_category", "_tutorialName"];
+
+private _key = [_category, _tutorialName];
+
+if (_key in vgm_c_tutorial_seenTutorials) exitWith {};
```
Comment (veteran29) [inline game/functions/systems/tutorial/client/fn_tutorial_trigger.sqf:26]: This is good enough for now but I feel like we could filter out seen tutorials during mission init and not even create a hooks for them.

**Review by veteran29 — CHANGES_REQUESTED** [review]

File: game/functions/systems/tutorial/client/fn_tutorial_trigger.sqf (hunk 4 above)
Comment (Spoffy) [reply game/functions/systems/tutorial/client/fn_tutorial_trigger.sqf:26]: Agreed

File: game/functions/systems/tutorial/client/fn_tutorial_preInit.sqf (hunk 3 above)
Comment (Spoffy) [reply game/functions/systems/tutorial/client/fn_tutorial_preInit.sqf:28]: That seems reasonable. Do these handle being in UIs, menus, correctly?

Comment (veteran29) [reply game/functions/systems/tutorial/client/fn_tutorial_preInit.sqf:28]: It seems to work only in main display and not on dialogs. Could use `actionKeysEx` to get the default value of the bind but I realized it's `H` by default which would collide with medical menu, so disregard this comment.

**Review by veteran29 — APPROVED** [review]

**Pull request merged**